ensime_shared/ensime.py

```diff
@@ -584,9 +584,13 @@ def handle_completion_info_list(self, call_id, payload):
         """Handler for a completion response."""
         completions = payload["completions"]
         self.log("handle_completion_info_list: in")
-        self.suggestions = [completion_to_suggest(c) for c in completions]
+        self.suggestions = [completion_to_suggest(c, self.format_type_params()) for c in completions]
         self.log("handle_completion_info_list: {}".format(self.suggestions))
 
+    def format_type_params(self):
+        """NOP as it's not available for Server 1.0"""
+        return lambda _: ""
+
```

Comment:
I much prefer the factory approach. It's much clearer to an uninitiated Pythonista like myself.